### ISSUE TITLE
Gateways created by `istio manifest generate` do not have unique pod selectors

### ISSUE THREAD

**Issue by quinn** [istio/istio#23688]
istio version 1.5.2

```
$ istioctl version --remote
client version: 1.5.2
control plane version: 1.5.2
data plane version: 1.5.2 (3 proxies)
```

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T11:56:40Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.9-gke.24", GitCommit:"39e41a8d6b7221b901a95d3af358dea6994b4a40", GitTreeState:"clean", BuildDate:"2020-02-29T01:24:35Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}
```

given this operator:

```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: istio-operator
spec:
  components:
    ingressGateways:
    - enabled: true
      k8s:
        env:
        - name: ISTIO_META_ROUTER_MODE
          value: sni-dnat
        hpaSpec:
          maxReplicas: 5
          metrics:
          - resource:
              name: cpu
              targetAverageUtilization: 80
            type: Resource
          minReplicas: 1
          scaleTargetRef:
            apiVersion: apps/v1
            kind: Deployment
            name: public-ingress
        resources:
          limits:
            cpu: 2000m
            memory: 1024Mi
          requests:
            cpu: 100m
            memory: 128Mi
        service:
          ports:
          - name: status-port
            port: 15020
            targetPort: 15020
          - name: http2
            port: 80
            targetPort: 80
          - name: https
            port: 443
          - name: kiali
            port: 15029
            targetPort: 15029
          - name: prometheus
            port: 15030
            targetPort: 15030
          - name: grafana
            port: 15031
            targetPort: 15031
          - name: tracing
            port: 15032
            targetPort: 15032
          - name: tls
            port: 15443
            targetPort: 15443
          - name: tcp
            port: 31400
        strategy:
          rollingUpdate:
            maxSurge: 100%
            maxUnavailable: 25%
      name: public-ingress
      label:
        app: public-ingress
        istio: ingressgateway
    - enabled: true
      k8s:
        env:
        - name: ISTIO_META_ROUTER_MODE
          value: sni-dnat
        hpaSpec:
          maxReplicas: 5
          metrics:
          - resource:
              name: cpu
              targetAverageUtilization: 80
            type: Resource
          minReplicas: 1
          scaleTargetRef:
            apiVersion: apps/v1
            kind: Deployment
            name: iap-ingress
        resources:
          limits:
            cpu: 2000m
            memory: 1024Mi
          requests:
            cpu: 100m
            memory: 128Mi
        service:
          ports:
          - name: status-port
            port: 15020
            targetPort: 15020
          - name: http2
            port: 80
            targetPort: 80
          - name: https
            port: 443
          - name: kiali
            port: 15029
            targetPort: 15029
          - name: prometheus
            port: 15030
            targetPort: 15030
          - name: grafana
            port: 15031
            targetPort: 15031
          - name: tracing
            port: 15032
            targetPort: 15032
          - name: tls
            port: 15443
            targetPort: 15443
          - name: tcp
            port: 31400
        strategy:
          rollingUpdate:
            maxSurge: 100%
            maxUnavailable: 25%
      name: iap-ingress
      label:
        app: iap-ingress
        istio: ingressgateway

```

All of the selectors use `app: istio-ingressgateway` rather than the name given here, so the pods in the two created deployments are not unique to the gateways. 
![image](https://user-images.githubusercontent.com/3764/81497011-b3075b00-9289-11ea-8bf1-f367e7da0d44.png)

**Comment by quinn** [comment]
It seems like what i was looking for was "labels" in GatewaySpec.

**Comment by quinn** [comment]
That didn't seem to work. I updated the original manifest in the issue description to match. Here is the template I'm referring to: 

https://github.com/istio/istio/blob/master/manifests/charts/gateways/istio-ingress/templates/deployment.yaml

To me, it looks like it would use the labels from the GatewaySpec (though the field there is labels, not label..)

**Comment by quinn** [comment]
Ok, I think I found it... in this file:

https://github.com/istio/istio/blob/master/manifests/charts/gateways/istio-ingress/values.yaml

The labels are in there. Can i somehow reference a different `values.yaml` for the `istio manifest generate` command?

**Comment by morvencao** [comment]
You can change `labels` field to other value for your `values.yaml` if you are intent to deploy extra gateway.

**Comment by quinn** [comment]
@morvencao that sounds great. thats actually my question: "Can i somehow reference a different values.yaml for the istio manifest generate command?"

**Comment by rytswd** [comment]
If you want to handle additional Helm `values.yaml` input in your IstioOperator, you can add `values` stanza as explained here https://istio.io/docs/reference/config/istio.operator.v1alpha1/#IstioOperatorSpec.
You can check out this example IstioOperator definition which has similar setup.
https://github.com/istio/istio/blob/master/manifests/examples/multicluster/values-istio-multicluster-gateways.yaml

As to the other point you raised regarding `app: istio-ingressgateway` label, I have raised an issue https://github.com/istio/istio/issues/23303 and found out that the fix is pending on the new release (for both v1.6 and v1.5).
There are a few other potential concerns with multiple gateway handling which I detailed more in the aforementioned issue. It will likely be shipped with v1.6.1.
